src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/getParams.tsx

```diff
@@ -103,7 +107,10 @@ interface Params {
 // Accepts `period` and `statsPeriod` but will only return `statsPeriod`
 //
 // TODO(billy): Make period parameter name consistent
-export function getParams(params: Params): {[key: string]: string | string[]} {
+export function getParams(
+  params: Params,
+  {allowEmptyPeriod = false}: {allowEmptyPeriod?: boolean} = {}
+) {
```

Comment:
I played with the implementation based on your changes, and this seems to produce a consistent return type.

```js
export function getParams(
  params: Params,
  {allowEmptyPeriod = false}: {allowEmptyPeriod?: boolean} = {}
): {
  [K in keyof Params]: NonNullable<Params[K]>;
} {
  const {start, end, period, statsPeriod, utc, ...otherParams} = params;

  // `statsPeriod` takes precendence for now
  let coercedPeriod = getStatsPeriodValue(statsPeriod) || getStatsPeriodValue(period);

  const dateTimeStart = getDateTimeString(start);
  const dateTimeEnd = getDateTimeString(end);

  if (!(dateTimeStart && dateTimeEnd)) {
    if (!coercedPeriod && !allowEmptyPeriod) {
      coercedPeriod = DEFAULT_STATS_PERIOD;
    }
  }

  const entries = Object.entries<string | string[] | null | undefined>({
    statsPeriod: coercedPeriod,
    start: coercedPeriod ? null : dateTimeStart,
    end: coercedPeriod ? null : dateTimeEnd,
    // coerce utc into a string (it can be both: a string representation from router,
    // or a boolean from time range picker)
    utc: getUtcValue(utc),
    ...otherParams,
  })
    // Filter null values
    .filter(([_key, value]: [string, string | string[] | null | undefined]) =>
      defined(value)
    ) as [keyof Params, string | string[]][];

  return Object.fromEntries<string | string[]>(entries);
}
```

Reply:
Hmm I see what you mean, we lose the keys of `Params` here

Reply:
I didn't realize this previously, but since we use `filter()` at runtime, we're unable to implicitly infer that `null | undefined` are dropped types from the value types of `Params`. 

An explicit type cast is necessary.